### DISCUSSION TITLE
When multiple users push data to Realtime Database to get the firebase-generated-key? How do they get their key and not someone else's?

### DISCUSSION THREAD

**Discussion by shafnaz** [GodotNuts/GodotFirebase#361] [Q&A] [answered]
Hello devs,

I have a question.
I see that Firebase generates their own key and keeps your data under that key.
![image](https://github.com/GodotNuts/GodotFirebase/assets/37254287/13ea238c-3ebb-4366-a6e5-49576e6ab1ad)

So once anyone is connected to this database, they all receive updates to the database.
I have this code to push data to realtime database once a new player registers. I would then store the firebase generated key in the player's firestore doc. But what if multiple new users are also pushing data to the database, if they are connected, could they receive the wrong data and store that firebase-generated-key?

```

func add_userboard1_to_database():
	globals.print_msg_server(["realtime. adding userboard1 to database"])
	var db : FirebaseDatabaseReference = Firebase.Database.get_database_reference("/active_boards", {})
	db.new_data_update.connect(_on_new_userboard1_update)
	db.push_successful.connect(_on_push_userboard1_successful)
	db.push_failed.connect(_on_push_userboard1_failed)
	var data = 	{
		ud.get_userboard1_id():	{
									"group1":	{
													"player1":	{
																"seat" : SEAT_OPEN,
																"pawn1" :	{"t":54, "s":-1},
																"pawn2" :	{"t":54, "s":-1},
																"pawn3" :	{"t":54, "s":-1},
																},
													"player2":	{
																"seat" : SEAT_OPEN,
																"pawn1" :	{"t":54, "s":-1},
																"pawn2" :	{"t":54, "s":-1},
																"pawn3" :	{"t":54, "s":-1},
																},
													"player3":	{
																"seat" : SEAT_OPEN,
																"pawn1" :	{"t":54, "s":-1},
																"pawn2" :	{"t":54, "s":-1},
																"pawn3" :	{"t":54, "s":-1},
																},
												}
								}
				}
	db.push(data)


func _on_new_userboard1_update(data):
	globals.print_msg_server(["new userboard1 data pushed. storing firebase-generated-key", data])
	ud.set_userboard1_firebase_key(data["key"])
	emit_signal("result", null)
```

Thank you

**Comment by WolfgangSenff** [comment, marked as answer]
They will definitely receive all of it. The way I typically solve this type of problem is by ensuring the data includes or even is keyed directly to the user's ID, so having like a document that has the user's ID as parse of the identifier, and then when the realtime database has data pushed, including the ID of the user that pushed it.

**Reply by shafnaz** [reply]
So I have to wait for that user id key, and then store it?

**Reply by WolfgangSenff** [reply]
I mean you should contain it in the request somewhere, and then anything that triggers off that should only do so if the user id matches what came back from auth. Not sure why you're waiting for anything.

**Reply by shafnaz** [reply]
Not waiting, I mean matching the key like you said. Alright! Got it!
